app/Cmd.hs

```diff
@@ -0,0 +1,71 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Cmd (
+  CmdOpts (CmdOpts),
+  MongoOpts (MongoOpts)
+  ) where
+
+import           Data.Bifunctor
+import           Data.Maybe
+import           Data.Text        hiding (group)
+import           Database.MongoDB hiding (Group, group)
+import           Options
+
+-------------------------------------------------------------------------------------------------------------------------
+-- Global options
+-------------------------------------------------------------------------------------------------------------------------
+data CmdOpts = CmdOpts {
+  mongoOpts :: MongoOpts,
+  port      :: Int
+  } deriving (Eq, Show)
+
+instance Options CmdOpts where
+  defineOptions = pure CmdOpts
+                  <*> defineOptions
+                  <*> simpleOption "port" 3000 "Port on which to listen for incoming HTTP calls."
```

Comment:
It's more common to see `f <$> x <*> y` than `pure f <*> x <*> y`. `(<$>) = fmap`

Reply:
Understood. I liked the "simplicity" of having the same operator for each step, but if it's not idiomatic, then I need to change it.

Reply:
Should be fixed in 24e7bde2bcc268d64f3c32bbab00daa4b263af27.